tracker/dmlc_tracker/ssh.py

```diff
@@ -19,25 +20,71 @@ def sync_dir(local_dir, slave_node, slave_dir):
     logging.info('rsync %s -> %s', local_dir, remote)
     prog = 'rsync -az --rsh="ssh -o StrictHostKeyChecking=no -p %s" %s %s' % (
         slave_node[1], local_dir, remote)
-    subprocess.check_call([prog], shell = True)
+    subprocess.check_call([prog], shell=True)
+
+def prepare_envs(args):
+    """
+    Load environment variables from arguments
+    """
+    envs = {}
+    # default env variables which are always passed by the system
+    envs['default'] = {'OMP_NUM_THREADS', 'KMP_AFFINITY', 'LD_LIBRARY_PATH', 'AWS_ACCESS_KEY_ID',
+                          'AWS_SECRET_ACCESS_KEY', 'DMLC_INTERFACE'}
+
+    # given by user with the option --env
+    envs['user'] = set([item for item in args.env.split(',') if item])
+    # remove those specified by user from default so we can confirm that user has set these vars
+    envs['default'].difference_update(envs['user'])
+    envs['server'] = parse_env_pairs(args.env_server)
+    envs['worker'] = parse_env_pairs(args.env_worker)
+    return envs
+
+def get_envs_str(dmlc_envs, addnl_envs, is_server):
+    """
+    Returns a string containing the command to export all relevant environment variables
+    :param dmlc_envs: required environment variables for distributed training
+    :param addnl_envs: environment variables which user might want and those specified explicitly
+    :param is_server: whether these variables are for a server node
+    :return: the export command string
+    """
+    # appends a pair of key and value to the list representing export command
+    def append_env_var(envs_list, key, value):
+        envs_list.append('export ' + key + '=' + str(value) + ';')
+
+    # add role to dmlc_envs
+    dmlc_envs['DMLC_ROLE'] = 'server' if is_server else 'worker'
 
-def get_env(pass_envs):
     envs = []
-    # get system envs
-    keys = ['OMP_NUM_THREADS', 'KMP_AFFINITY', 'LD_LIBRARY_PATH', 'AWS_ACCESS_KEY_ID',
-            'AWS_SECRET_ACCESS_KEY', 'DMLC_INTERFACE']
-    for k in keys:
+    for k in addnl_envs['default']:
+        v = os.getenv(k)
+        if v is not None:
+            append_env_var(envs, k, v)
+
+    for k in addnl_envs['user']:
         v = os.getenv(k)
```

Comment:
if user has already supplied the key value in arguments, does it make sense to check os environment variables here ?
I think this can also break, please verify what happens if 
- I give --env VIKAS:1 
And Vikas is not in my environment variable, will it throw exception - "The environment variable '+ k + ' was passed but not set"